rep-0152.rst

```diff
@@ -0,0 +1,144 @@
+REP: 152
+Title: Orphaned Pakcage Maintaining
+Author: Kei Okada
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 27-Dec-2017
+
+Outline
+=======
+
+#. Abstract_
+#. Motivation_
+#. Rationale_
+#. Operations_
+#. References_
+#. Copyright_
+
+Abstract
+========
+
+This REP specifies how we can organize and release packages without active maintainers.
+
+Motivation
+==========
+
+As the size of ROS ecosystem grows, there are several packages which loosing active maintainers.
+Main drawback of this situation is these packages do not get released into newer ROS distro coming every year, even if they are commonly used (For example, usb camera drivers are not released into Kinetic for more than half year [1]_ [2]_).
+Ideally everyone willing to use newer ROS distro almost equivalent to the set of packages in previous ROS distro.
+
+It is understandable that original maintainer could not continue serving as an active maintainer, because of starting new project, moving an institute/company, loosing interest and so on.
+And being a maintainer is not a easy job. It includes respond to incoming questions and issues, check for proposed changes and decide to include them to a main source tree and so on.
+
+The idea behind this document is to have a team of volunteers who can do ONLY the release job.
+This will enable to continue using same set of previous ROS distros with minimum set of community efforts.
+To this end, we followed Debian QA Group System and Debian Developer's Reference ('Dealing with inactive and/or unreachable maintainers'[3]_), which is considered non-normative, but best practice.
+
+In ROS system, we use 'Unmaintained' package status([4]_) and ROS Orphaned Packages Group([5]_), who do not have to take full responsibility to package maintenance jobs, but expecting to take minimum responsibilities such as just to releasing into newer ROS distros.
+
+
+Rationale
+=========
+
+REP-0140 [6]_ and REP-0141 [7]_ provide maintainer information in package manifest file and maintainer status in ROS distribution file.
+
+Maintenance Guide in ROSWiki [8]_ provides how to claim maintainership for orphaned packages.
+
+Operations
+==========
+
+1-A. Orphaning a package by maintainers
```

Comment:
With the current proposal I don't see a way to distinguish the following two cases:

* The group of orphaned package maintainers "chooses" that a specific package is orphaned, updates the manifest accordingly and starts doing releases of that package
* A maintainer who can no longer maintain a package updates the manifest accordingly.

From a users point of view it would be nice to be able to distinguish these two cases. In the first case there is the expectation that there will be releases in the future. In the second case likely not (until the orphaned package maintainers actually "accept" the work and start doing releases for that package).

Reply:
@dirk-thomas thank you for comments,  i have fixed most of comment. On this comment, `A. Orphaned a package by maintainer` corresponds to 
`A maintainer who can no longer maintain a package updates the manifest accordingly`.
 and if some of maintainer want to maintain the package by the orphaned team, I expect they choose `Orphaning a package by users`, not as a member of the team, but as an individuals.

To minimize the effort of orphaned maintainers, I intentionally remove the process of something like `the group of orphaned package maintainers "chooses" `So in both cases, user might be not sure if the packages will be releases in the future.  To put such responsibility to the orphaned team feel to much for me. How do you think?

Reply:
So for all packages marked as "orphaned" in the manifest files there is no information which ones are going to be released by the group, right? I think it would be valuable for users to differentiate them but I understand your rational why the group doesn't want to "promise" that.

Reply:
Yeah, my first thought on this, is to create the organization like `ros-unmaintained` and once the package is marked as orphaned, then we transfer the package to that organization. If so everyone can see who is going to (or able to ) release these package, by looking at the "users" or "teams" of the organization. 
But the draw back of this idea is that the location of the package will be moving around... If we can create github teams across different organization, then it would be great.
c.f. https://discourse.ros.org/t/releasing-repositories-form-other-people/1797/11?u=k-okada

Reply:
> the draw back of this idea is that the location of the package will be moving around

:-1: on moving repos around just to indicate their status. It generates quite some friction which is unnecessary.